pyccel/codegen/printing/ccode.py

```diff
@@ -553,6 +554,17 @@ def _print_PythonBool(self, expr):
     def _print_Literal(self, expr):
         return repr(expr.python_value)
 
+    def _print_LiteralInteger(self, expr):
+        if isinstance(expr, LiteralInteger) and get_final_precision(expr) == 8:
+            self.add_import(c_imports['stdint'])
+            return f"INT64_C({repr(expr.python_value)})"
+        return repr(expr.python_value)
```

Comment:
Do we support `int8` or `int16`? They have numpy conversion functions, but I don't remember if we have a way to print them. I think there may be a related issue.

Reply:
the support for `int16` is fine as the `int16` and `int8` literals don't need a prefix and when printing `int16` the format is correct (`%hd`) but I just notice that the `int8` is printed as char (`%c`) [here](https://github.com/pyccel/pyccel/blob/02aba7c92344a30e28206b6abea6f13efac148cb/pyccel/codegen/printing/ccode.py#L803) I wonder if there is any reason for it

Reply:
normally it should be `%hhd`

Reply:
Looking at [this discussion](https://stackoverflow.com/questions/13052880/format-specifier-of-int8-data-type-in-c) I don't know to what degree `%hhd` is specified in the standard, which may explain how we ended up with `%c`. As they say though, the safest is to always use `inttypes.h` to be sure that there are no problems on strange machines

Reply:
okay if that's the case we can use `PRIi8` and `PRIi16` for `int8` and `int16` for more portability as we did support macros anyways.

Reply:
in my macOS the `PRId8` expanded to `%hhd` but on my Linux machine it's just `%d`